spec/controllers/provision_controller_spec.rb

```diff
@@ -1,7 +1,129 @@
 require 'spec_helper'
 
-class ProvisionControllerTest < ActionDispatch::IntegrationTest
-  # spec "the truth" do
-  #   assert true
-  # end
+require 'spec_helper'
+
+RSpec.describe ProvisionController, type: :request do
+  let(:username) { ENV['SECURITY_USER_NAME'] }
+  let(:password) { ENV['SECURITY_USER_PASSWORD'] }
+
+  let(:headers) do
+    {
+      'X-Broker-API-Version' => '2.13',
+      'HTTP_AUTHORIZATION' => ActionController::HttpAuthentication::Basic.encode_credentials(username, password)
+    }
+  end
+
+  describe 'PUT' do
+    let(:legacy_params) do
+      {
+        service_id: 'c024e536-6dc4-45c6-8a53-127e7f8275ab',
+        plan_id: '3a116ac2-fc8b-496f-a715-e9a1b205d05c.community',
+        context: {
+          platform: 'cloudfoundry'
+        },
+        organization_guid: 'test_org',
+        space_guid: 'test_space'
+      }
+    end
+  
+    let(:params) do 
+      legacy_params.merge({
+        context: {
```

Comment:
are you also getting the org/space from context on provision? I expected we would just get them from the org / space GUID passed in the request, since that's required as per the spec

Reply:
This is the same as above, I read the spec that I should be using `context` instead of the values directly in the request.

What do you think about adding a fallback for reading the values directly from `params` if they don't exist in `context`?

I didn't do this before since we've talked about the new capabilities being only for PCF 2.0+ where the `context` will exist. I don't think it makes sense to use only the param-direct values, since they will be going away in the future.

What do you think?

Reply:
after rereading, I think this will be fine as-is - but I am double checking with Pivotal to ensure `context` is supported in PAS 2.0+ just in case